troi/tools/utils.py

```diff
@@ -18,3 +21,142 @@ def create_http_session():
+        self.access_token = access_token
+        self.headers = {
+            "Authorization": f"OAuth {self.access_token}",
+            "Content-Type": "application/json"
+        }
+        self.session = self._create_http_session()
+
+    def _create_http_session(self):
+        """ Create an HTTP session with retry strategy for handling rate limits and server errors.
+        """
+        retry_strategy = Retry(
+            total=3,
+            status_forcelist=[429, 500, 502, 503, 504], 
+            allowed_methods=["HEAD", "GET", "OPTIONS"], 
+            backoff_factor=1
+        )
+
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        http = requests.Session()
+        http.mount("https://", adapter)
+        http.mount("http://", adapter)
+
+        def raise_for_status_hook(response, *args, **kwargs):
+            try:
+                response.raise_for_status()
+            except HTTPError as http_err:
+                print(f"HTTP error occurred: {http_err}")
+                raise
+
+        http.hooks["response"] = [raise_for_status_hook]
+        return http
+
+    def create_playlist(self, title, sharing="public" ,track_ids=None, description=None):
+        url = f"{SOUNDCLOUD_URL}/playlists"
+        data = {
+            "playlist": {
+                "title": title,
+                "sharing": sharing
+            }
+        }
+        if description:
+            data["playlist"]["description"] = description
+        response = self.session.post(url, headers=self.headers, data=json.dumps(data))
+        return response.json()
+
+    def add_playlist_tracks(self, playlist_id, track_ids):
+        url = f"{SOUNDCLOUD_URL}/playlists/{playlist_id}"
+        data = {
+            "playlist": {
+                "tracks": [{"id": track_id} for track_id in track_ids]
+            }
+        }
+        response = self.session.put(url, headers=self.headers, data=json.dumps(data))
+        return response.json()
+
+    def update_playlist_details(self, playlist_id, title=None, description=None):
+        url = f"{SOUNDCLOUD_URL}/playlists/{playlist_id}"
+        data = {
+            "playlist": {}
+        }
+        if title:
+            data["playlist"]["title"] = title
+        if description:
+            data["playlist"]["description"] = description
+
+        response = self.session.put(url, headers=self.headers, data=json.dumps(data))
+        return response.json()
+    
+    def get_track_details(self, track_ids, **kwargs):
+        track_details = []
+
+        for track_id in track_ids:
+            url = f"{SOUNDCLOUD_URL}/tracks/{track_id}"
+            response = self.session.get(url, headers=self.headers, params=kwargs)
+
+            if response.status_code != 200:
+                track_details.append({
+                    "id": track_id,
+                    "title": None,
+                    "playable": False
+                })
+                continue
+
+            data = response.json()
```

Comment:
This looks wrong, track_details endpoint does not return a collection. You should create the track_data here directly from the response without any loop.